### PR TITLE
Add partition sort algorithm

### PR DESCRIPTION
**Pull request by danghai**
Hi,
I make PR for adding partition sort algorithm.

### CODE REVIEW

**Comment by yanglbme** [comment]
Hi @danghai, the partition sort algorithm already exists in the repo. See [QuickSort.c](https://github.com/TheAlgorithms/C/blob/master/sorting/QuickSort.c).

**Comment by danghai** [comment]
Hi @yanglbme , it is a different way for quickSort.

File: sorting/partitionSort.c [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+void printArray(int arr[], int n) {
+    int i;
+	for(i = 0; i < n; i++)
+	    printf("%d ", arr[i]);
+	printf("\n");
+}
+
+int main() {
+    int arr[20];
+	int i, range = 100;
```
Comment (yanglbme) [inline sorting/partitionSort.c]: please format your code before raising a PR.

```c
#include <stdlib.h>
#include <stdio.h>

void swap(int *a, int *b)
{
    int tmp = *a;
    *a = *b;
    *b = tmp;
}

int partition(int arr[], int low, int high)
{
    int pivot = arr[low];
    int i = low - 1, j = high + 1;

    while (1)
    {
        /* Find leftmost element >= pivot */
        do
        {
            i++;
        } while (arr[i] < pivot);

        /* Find rightmost element <= pivot */
        do
        {
            j--;
        } while (arr[j] > pivot);

        /* if two pointers met */
        if (i >= j)
            return j;

        swap(&arr[i], &arr[j]);
    }
}

void partitionSort(int arr[], int low, int high)
{
    if (low < high)
    {
        int value = partition(arr, low, high);

        partitionSort(arr, low, value);
        partitionSort(arr, value + 1, high);
    }
}

void printArray(int arr[], int n)
{
    int i;
    for (i = 0; i < n; i++)
        printf("%d ", arr[i]);
    printf("\n");
}

int main()
{
    int arr[20];
    int i, range = 100;

    for (i = 0; i < 20; i++)
    {
        arr[i] = rand() % range + 1;
    }
    int size = sizeof arr / sizeof arr[0];

    printf("Array: \n");
    printArray(arr, size);
    partitionSort(arr, 0, size - 1);
    printf("Sorted Array: \n");
    printArray(arr, size);

    return 0;
}

```

**Comment by danghai** [comment]
@yanglbme I fixed it

**Pull request merged**